### ISSUE TITLE
Misplaced letters

### ISSUE THREAD

**Issue by jbezos** [lualatex/luaotfload#423]
There are misplaced letters in the following example, because some internal kerns are inserted at the wrong places. It seems it has been fixed in ConTeXt, but unfortutately the context loader doesn't work for me. Are there plans to update luaotfload?

```
\documentclass{book}
\usepackage{fontspec}
\setmainfont[Script=Arabic]{FreeSerif}

\begin{document}
\hsize 3.5cm

\pardir TRT
\textdir TRT

1234 الإشارات برهن

\mbox{1234 الإشارات برهن}

\halign{#\cr 1234 الإشارات برهن\cr}

\end{document}
```

**Comment by u-fischer** [comment]
With an experimental setup (newest fontloader from context loaded with a luaotfload.conf and some patches luaotfload files you can find at https://github.com/u-fischer/lua-font-pond/tree/master/texmf/tex/luatex my output looks like this:

![image](https://user-images.githubusercontent.com/4047173/43212120-78103544-9033-11e8-9ec6-f082c2cb0703.png)

**Comment by u-fischer** [comment]
Imho this has been resolved with luaotfload 2.9/2.91. If not feel free to reopen the issue.
